src/pages/events/EventShow.jsx

```diff
@@ -7,265 +7,321 @@ import "./Events.scss";
 import EventSignIn from "./EventSignIn";
 
 export default function EventShow() {
+  const [isLoading, setIsLoading] = useState(true);
   const { path, url } = useRouteMatch();
   const { event_id } = useParams();
-  const event = {
-    "id": event_id,
-    "name": `Event ${event_id}`,
-    "format": "NLHE",
-    "start_date": new Date("2020-02-01T19:00:00"),
-    "notes": "First testing",
-    "count": 3,
-    "state": 0
+  const [event, setEvent] = useState({
+    id: event_id,
+    name: "",
+    start_date: "",
+    format: "",
+    notes: "",
+    semester_id: "",
+    state: 0
+  });
+  const [participants, setParticipants] = useState([]);
+  const [filteredParticipants, setFilteredParticipants] = useState([]);
+  const [search, setSearch] = useState("");
+
+  const [error, setError] = useState("");
+
+  const searchParticipants = (e) => {
+    setSearch(e.target.value);
+    e.preventDefault();
+
+    setFilteredParticipants(participants.filter((participant) => 
+      participant.first_name.toLowerCase().includes(search)
+      || participant.last_name.toLowerCase().includes(search)

```

Comment:
The reason the search takes a while is you are searching the entire array twice, An easier solution is to do something like I did on members, where you combine the first name and last name to a string and do a regex search